jobs/export_process/worker.go

```diff
@@ -44,15 +45,21 @@ func MakeWorker(jobServer *jobs.JobServer, app AppIface) model.Worker {
 
 		errCh := make(chan *model.AppError, 1)
 		go func() {
-			defer close(errCh)
+			defer func() {
+				close(errCh)
+
+				// we close the reader here to prevent a deadlock when the bulk exporter tries to
+				// write into the pipe while app.WriteFile has already returned. The error will be
+				// returned by the writer part of the pipe when app.BulkExport tries to call
+				// wr.Write() on it.
+				rd.CloseWithError(errors.New("Worker: writer has been closed"))
+			}()
 			_, appErr := app.WriteFile(rd, filepath.Join(outPath, exportFilename))
 			errCh <- appErr
 		}()
 
 		appErr := app.BulkExport(request.EmptyContext(app.Log()), wr, outPath, opts)
-		if err := wr.Close(); err != nil {
-			mlog.Warn("Worker: error closing writer")
-		}
+		wr.Close()
```

Comment:
There is no error returned by this `Close`. I removed the check because it will always return `nil` (I know, there are different opinions about this). 1/5

Reply:
Ah okay, thanks. Maybe just add a comment for readers to be aware that this is intended?